ownCloud/ownCloud/Bookmarks/Connection Issues /IssuesViewController.swift

```diff
@@ -0,0 +1,217 @@
+import ownCloudSDK
+import ownCloudUI
+
+class IssuesViewController: UIViewController {
+
+    internal let issuesTableView: UITableView = UITableView()
+    public let bottomContainer: UIView = UIView()
+    private var issues: [OCConnectionIssue] = []
+    private var tableHeighConstraint: NSLayoutConstraint?
+    private var headerTitle: String?
+    private var completionHandler: (() -> Void)?
+
+    private var modalPresentationVC: UIViewControllerTransitioningDelegate?
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.setupTransitions()
+    }
+
+    init(issues: [OCConnectionIssue]?, headerTitle: String?, completionHandler: (() -> Void)? = nil) {
+        super.init(nibName: nil, bundle: nil)
+        self.headerTitle = headerTitle
+        self.completionHandler = completionHandler
+        if issues != nil {
+            self.issues = issues!
+        }
+        self.setupTransitions()
+    }
+
+    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
+        super.dismiss(animated: flag, completion: completion)
+
+        if let completionHandler = completionHandler {
+            completionHandler()
+        }
+    }
+
+    private func setupTransitions() {
+        self.modalPresentationStyle = .custom
+        let transitioningDLG = IssuesTransitioningDelegate()
+        self.modalPresentationVC = transitioningDLG
+        self.transitioningDelegate = transitioningDLG
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        issuesTableView.delegate = self
+        issuesTableView.dataSource = self
+
+        // Do any additional setup after loading the view.
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        bottomContainer.translatesAutoresizingMaskIntoConstraints = false
+        self.view.addSubview(bottomContainer)
+
+        bottomContainer.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor, constant: -20).isActive = true
+        bottomContainer.leftAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.leftAnchor, constant: 20).isActive = true
+        bottomContainer.rightAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.rightAnchor, constant: -20).isActive = true
+        bottomContainer.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        bottomContainer.contentHuggingPriority(for: .vertical)
+
+        issuesTableView.translatesAutoresizingMaskIntoConstraints = false
+        issuesTableView.layer.cornerRadius = 10
+        issuesTableView.backgroundColor = UIColor(hex: 0xF2F2F2)
+        issuesTableView.bounces = false
+        issuesTableView.showsVerticalScrollIndicator = false
+
```

Comment:
If more than one issue is presented, the separator is not filling the entire width.
![bildschirmfoto 2018-03-22 um 23 30 49](https://user-images.githubusercontent.com/639669/37802027-17135ff8-2e29-11e8-9949-cf13b37f3010.png)

You can fix this by setting issuesTableView.separatorInset to zero values.

Reply:
Done!